### PR TITLE
Continue testing react v17 with e2e tests

### PR DESCRIPTION
**Pull request by ijjk**
Follow-up to https://github.com/vercel/next.js/pull/35774 this continues testing react v17 as well as react 18 in our e2e tests to ensure coverage since we have the concurrency to accommodate this.

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 48d45b4e8d24bf76b232a259d7822307c098f5b6

`yarn testheadless test/development/basic/hmr.test.ts` 
- basic HMR > Error Recovery > should recover after a bad return from the render function

<details>
<summary>Expand output</summary>

  ● basic HMR › Error Recovery › should recover after a bad return from the render function

    expect(received).toMatch(expected)

    Expected substring: "Objects are not valid as a React child (found: [object RegExp]). If you meant to render a collection of children, use an array instead."
    Received string:    " 1 of 1 unhandled error
    Server Error·
    Error: Objects are not valid as a React child (found: /search/). If you meant to render a collection of children, use an array instead.·
    This error happened while generating the page. Any console logs will be displayed in the terminal window."

      593 |             'Unknown'
      594 |           )
    > 595 |         ).toMatch(
          |           ^
      596 |           'Objects are not valid as a React child (found: [object RegExp]). If you meant to render a collection of children, use an array instead.'
      597 |         )
      598 |

      at Object.<anonymous> (development/basic/hmr.test.ts:595:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by huozhi** [comment]
looks like the only difference between 17 and 18 for `should recover after a bad return from the render function` is the stringified name of regex, it's either regex string itself, or the stringified of regex constructor..

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 62b92efea6dd951bb86c184509b6bbcaca0c20d5

`yarn testheadless test/development/basic/hmr.test.ts` 
- basic HMR > Error Recovery > should recover after a bad return from the render function

<details>
<summary>Expand output</summary>

  ● basic HMR › Error Recovery › should recover after a bad return from the render function

    expect(received).toMatch(expected)

    Expected substring: "Objects are not valid as a React child (found: [object RegExp]). If you meant to render a collection of children, use an array instead."
    Received string:    " 1 of 1 unhandled error
    Server Error·
    Error: Objects are not valid as a React child (found: /search/). If you meant to render a collection of children, use an array instead.·
    This error happened while generating the page. Any console logs will be displayed in the terminal window."

      595 |             'Unknown'
      596 |           )
    > 597 |         ).toMatch(
          |           ^
      598 |           `Objects are not valid as a React child (found: ${
      599 |             isReact17 ? '/search/' : '[object RegExp]'
      600 |           }). If you meant to render a collection of children, use an array instead.`

      at Object.<anonymous> (development/basic/hmr.test.ts:597:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 62b92efea6dd951bb86c184509b6bbcaca0c20d5

`yarn testheadless test/development/basic/hmr.test.ts` 
- basic HMR > Error Recovery > should recover after a bad return from the render function

<details>
<summary>Expand output</summary>

  ● basic HMR › Error Recovery › should recover after a bad return from the render function

    expect(received).toMatch(expected)

    Expected substring: "Objects are not valid as a React child (found: [object RegExp]). If you meant to render a collection of children, use an array instead."
    Received string:    " 1 of 1 unhandled error
    Server Error·
    Error: Objects are not valid as a React child (found: /search/). If you meant to render a collection of children, use an array instead.·
    This error happened while generating the page. Any console logs will be displayed in the terminal window."

      595 |             'Unknown'
      596 |           )
    > 597 |         ).toMatch(
          |           ^
      598 |           `Objects are not valid as a React child (found: ${
      599 |             isReact17 ? '/search/' : '[object RegExp]'
      600 |           }). If you meant to render a collection of children, use an array instead.`

      at Object.<anonymous> (development/basic/hmr.test.ts:597:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 496582ba5ff0586b4e9e71b15ce78045a1206f65

`yarn testheadless test/development/basic/hmr.test.ts` 
- basic HMR > Error Recovery > should recover after a bad return from the render function

<details>
<summary>Expand output</summary>

  ● basic HMR › Error Recovery › should recover after a bad return from the render function

    expect(received).toMatch(expected)

    Expected substring: "Objects are not valid as a React child (found: [object RegExp]). If you meant to render a collection of children, use an array instead."
    Received string:    " 1 of 1 unhandled error
    Server Error·
    Error: Objects are not valid as a React child (found: /search/). If you meant to render a collection of children, use an array instead.·
    This error happened while generating the page. Any console logs will be displayed in the terminal window."

      595 |             'Unknown'
      596 |           )
    > 597 |         ).toMatch(
          |           ^
      598 |           `Objects are not valid as a React child (found: ${
      599 |             isReact17 ? '/search/' : '[object RegExp]'
      600 |           }). If you meant to render a collection of children, use an array instead.`

      at Object.<anonymous> (development/basic/hmr.test.ts:597:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Review by huozhi — DISMISSED** [review]

**Comment by ijjk** [comment]
## Failing test suites 
Commit: ee15c9d5666ca4d2a24eb6ee6141d4ca1f367a59

`yarn testheadless test/development/basic-basepath/hmr.test.ts` 
- basic HMR > Error Recovery > should recover after a bad return from the render function

<details>
<summary>Expand output</summary>

  ● basic HMR › Error Recovery › should recover after a bad return from the render function

    expect(received).toMatch(expected)

    Expected substring: "Objects are not valid as a React child (found: /search/). If you meant to render a collection of children, use an array instead."
    Received string:    " 1 of 1 unhandled error
    Server Error·
    Error: Objects are not valid as a React child (found: [object RegExp]). If you meant to render a collection of children, use an array instead.·
    This error happened while generating the page. Any console logs will be displayed in the terminal window."

      540 |             'Unknown'
      541 |           )
    > 542 |         ).toMatch(
          |           ^
      543 |           'Objects are not valid as a React child (found: /search/). If you meant to render a collection of children, use an array instead.'
      544 |         )
      545 |

      at Object.<anonymous> (development/basic-basepath/hmr.test.ts:542:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
## Failing test suites 
Commit: ee15c9d5666ca4d2a24eb6ee6141d4ca1f367a59

`yarn testheadless test/development/basic-basepath/hmr.test.ts` 
- basic HMR > Error Recovery > should recover after a bad return from the render function

<details>
<summary>Expand output</summary>

  ● basic HMR › Error Recovery › should recover after a bad return from the render function

    expect(received).toMatch(expected)

    Expected substring: "Objects are not valid as a React child (found: /search/). If you meant to render a collection of children, use an array instead."
    Received string:    " 1 of 1 unhandled error
    Server Error·
    Error: Objects are not valid as a React child (found: [object RegExp]). If you meant to render a collection of children, use an array instead.·
    This error happened while generating the page. Any console logs will be displayed in the terminal window."

      540 |             'Unknown'
      541 |           )
    > 542 |         ).toMatch(
          |           ^
      543 |           'Objects are not valid as a React child (found: /search/). If you meant to render a collection of children, use an array instead.'
      544 |         )
      545 |

      at Object.<anonymous> (development/basic-basepath/hmr.test.ts:542:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| buildDuration | 20.2s | 20.4s | ⚠️ +159ms |
| buildDurationCached | 7.8s | 7.8s | ⚠️ +6ms |
| nodeModulesSize | 477 MB | 477 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 4.362 | 4.32 | -0.04 |
| / avg req/sec | 573.18 | 578.69 | +5.51 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.22 | 2.188 | -0.03 |
| /error-in-render avg req/sec | 1126.21 | 1142.58 | +16.37 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| 925.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42 kB | 42 kB | ✓ |
| main-HASH.js gzip | 28.4 kB | 28.4 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 72.1 kB | 72.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.36 kB | 1.36 kB | ✓ |
| _error-HASH.js gzip | 192 B | 192 B | ✓ |
| amp-HASH.js gzip | 309 B | 309 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 3.05 kB | 3.05 kB | ✓ |
| head-HASH.js gzip | 351 B | 351 B | ✓ |
| hooks-HASH.js gzip | 920 B | 920 B | ✓ |
| image-HASH.js gzip | 5.48 kB | 5.48 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.26 kB | 2.26 kB | ✓ |
| routerDirect..HASH.js gzip | 320 B | 320 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 319 B | 319 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 15.7 kB | 15.7 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 545 B | 545 B | ✓ |
| withRouter.html gzip | 525 B | 525 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| buildDuration | 23.8s | 24s | ⚠️ +183ms |
| buildDurationCached | 7.9s | 7.9s | -10ms |
| nodeModulesSize | 477 MB | 477 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 4.369 | 4.354 | -0.01 |
| / avg req/sec | 572.24 | 574.13 | +1.89 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.188 | 2.227 | ⚠️ +0.04 |
| /error-in-render avg req/sec | 1142.78 | 1122.43 | ⚠️ -20.35 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| 925.HASH.js gzip | 178 B | 178 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 28.8 kB | 28.8 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 72.7 kB | 72.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 179 B | 179 B | ✓ |
| amp-HASH.js gzip | 313 B | 313 B | ✓ |
| css-HASH.js gzip | 325 B | 325 B | ✓ |
| dynamic-HASH.js gzip | 3.03 kB | 3.03 kB | ✓ |
| head-HASH.js gzip | 351 B | 351 B | ✓ |
| hooks-HASH.js gzip | 921 B | 921 B | ✓ |
| image-HASH.js gzip | 5.59 kB | 5.59 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 2.33 kB | 2.33 kB | ✓ |
| routerDirect..HASH.js gzip | 322 B | 322 B | ✓ |
| script-HASH.js gzip | 388 B | 388 B | ✓ |
| withRouter-HASH.js gzip | 317 B | 317 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 15.8 kB | 15.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| index.html gzip | 532 B | 532 B | ✓ |
| link.html gzip | 546 B | 546 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
Commit: ee15c9d5666ca4d2a24eb6ee6141d4ca1f367a59

File: test/lib/next-modes/base.ts [hunk 1]
```diff
@@ -88,16 +88,22 @@ export class NextInstance {
...
-          ...this.dependencies,
-          ...((this.packageJson.dependencies as object | undefined) || {}),
-        },
+        finalDependencies,
         this.installCommand,
         this.packageJson
       )
+      console.log('Using react version', {
+        env: reactVersion,
+        actual: finalDependencies.react,
```
Comment (styfle) [inline test/lib/next-modes/base.ts:105]: These will always be equal right?

You might want to read package.json here to get "actual"

Comment (ijjk) [reply test/lib/next-modes/base.ts:105]: A specific test can override the react version by specifying it in the `dependencies` for `createNext` manually.

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 5760a9f8867213f95a9fac073fe27a9ec01b780f

`yarn testheadless test/development/basic-basepath/hmr.test.ts` 
- basic HMR > Error Recovery > should recover after a bad return from the render function

<details>
<summary>Expand output</summary>

  ● basic HMR › Error Recovery › should recover after a bad return from the render function

    expect(received).toMatch(expected)

    Expected substring: "Objects are not valid as a React child (found: /search/). If you meant to render a collection of children, use an array instead."
    Received string:    " 1 of 1 unhandled error
    Server Error·
    Error: Objects are not valid as a React child (found: [object RegExp]). If you meant to render a collection of children, use an array instead.·
    This error happened while generating the page. Any console logs will be displayed in the terminal window."

      540 |             'Unknown'
      541 |           )
    > 542 |         ).toMatch(
          |           ^
      543 |           'Objects are not valid as a React child (found: /search/). If you meant to render a collection of children, use an array instead.'
      544 |         )
      545 |

      at Object.<anonymous> (development/basic-basepath/hmr.test.ts:542:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
## Failing test suites 
Commit: d0820e447b1c71a3fb9fcfd923c06ea5249b06fb

`yarn testheadless test/e2e/basepath.test.ts` 
- basePath > should use urls with basepath in router events for cancelled routes

<details>
<summary>Expand output</summary>

  ● basePath › should use urls with basepath in router events for cancelled routes

    expect(received).toEqual(expected) // deep equality

    - Expected  - 0
    + Received  + 7

    @@ -1,7 +1,14 @@
      Array [
        Array [
    +     "routeChangeComplete",
    +     "/docs/hello",
    +     Object {
    +       "shallow": true,
    +     },
    +   ],
    +   Array [
          "routeChangeStart",
          "/docs/slow-route",
          Object {
            "shallow": false,
          },

      773 |
      774 |         const eventLog = await browser.eval('window._getEventLog()')
    > 775 |         expect(eventLog).toEqual([
          |                          ^
      776 |           ['routeChangeStart', `${basePath}/slow-route`, { shallow: false }],
      777 |           [
      778 |             'routeChangeError',

      at Object.<anonymous> (e2e/basepath.test.ts:775:26)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
## Failing test suites 
Commit: d4ee34304123d161af1c4689175782ee5a7e9182

`yarn testheadless test/e2e/basepath.test.ts` 
- basePath > should use urls with basepath in router events for failed route change

<details>
<summary>Expand output</summary>

  ● basePath › should use urls with basepath in router events for failed route change

    expect(received).toEqual(expected) // deep equality

    - Expected  - 0
    + Received  + 7

    @@ -1,7 +1,14 @@
      Array [
        Array [
    +     "routeChangeComplete",
    +     "/docs/hello",
    +     Object {
    +       "shallow": true,
    +     },
    +   ],
    +   Array [
          "routeChangeStart",
          "/docs/error-route",
          Object {
            "shallow": false,
          },

      800 |
      801 |         const eventLog = await browser.eval('window._getEventLog()')
    > 802 |         expect(eventLog).toEqual([
          |                          ^
      803 |           ['routeChangeStart', `${basePath}/error-route`, { shallow: false }],
      804 |           [
      805 |             'routeChangeError',

      at Object.<anonymous> (e2e/basepath.test.ts:802:26)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| buildDuration | 19.9s | 18.7s | -1.2s |
| buildDurationCached | 7.6s | 7.2s | -385ms |
| nodeModulesSize | 477 MB | 477 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.806 | 3.804 | 0 |
| / avg req/sec | 656.91 | 657.27 | +0.36 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.668 | 1.649 | -0.02 |
| /error-in-render avg req/sec | 1498.76 | 1516.41 | +17.65 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| 925.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42 kB | 42 kB | ✓ |
| main-HASH.js gzip | 28.4 kB | 28.4 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 72.1 kB | 72.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.36 kB | 1.36 kB | ✓ |
| _error-HASH.js gzip | 192 B | 192 B | ✓ |
| amp-HASH.js gzip | 309 B | 309 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 3.05 kB | 3.05 kB | ✓ |
| head-HASH.js gzip | 351 B | 351 B | ✓ |
| hooks-HASH.js gzip | 920 B | 920 B | ✓ |
| image-HASH.js gzip | 5.48 kB | 5.48 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.26 kB | 2.26 kB | ✓ |
| routerDirect..HASH.js gzip | 320 B | 320 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 319 B | 319 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 15.7 kB | 15.7 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 545 B | 545 B | ✓ |
| withRouter.html gzip | 525 B | 525 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

```diff
@@ -1505,14 +1505,14 @@
           },
           [callbacks]
         );
-        if (false) {
-        }
         // We should ask to measure the Web Vitals after rendering completes so we
         // don't cause any hydration delay:
         _react.default.useEffect(function() {
           (0, _performanceRelayer).default(onPerfEntry);
           (0, _vitals).flushBufferedVitalsMetrics();
         }, []);
+        if (false) {
+        }
         return children;
       }
       // Dummy component that we render as a child of Root so that we can
```
</details>
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-444bd3ddf1f35edc.js"
+      src="/_next/static/chunks/main-ca8f6d6aba194701.js"
       defer=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-444bd3ddf1f35edc.js"
+      src="/_next/static/chunks/main-ca8f6d6aba194701.js"
       defer=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-444bd3ddf1f35edc.js"
+      src="/_next/static/chunks/main-ca8f6d6aba194701.js"
       defer=""
     ></script>
     <script
```
</details>
</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| buildDuration | 21.5s | 22.2s | ⚠️ +683ms |
| buildDurationCached | 7.5s | 7.2s | -287ms |
| nodeModulesSize | 477 MB | 477 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.75 | 3.705 | -0.04 |
| / avg req/sec | 666.74 | 674.81 | +8.07 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.681 | 1.639 | -0.04 |
| /error-in-render avg req/sec | 1487.29 | 1525.59 | +38.3 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| 925.HASH.js gzip | 178 B | 178 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 28.8 kB | 28.8 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 72.7 kB | 72.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 179 B | 179 B | ✓ |
| amp-HASH.js gzip | 313 B | 313 B | ✓ |
| css-HASH.js gzip | 325 B | 325 B | ✓ |
| dynamic-HASH.js gzip | 3.03 kB | 3.03 kB | ✓ |
| head-HASH.js gzip | 351 B | 351 B | ✓ |
| hooks-HASH.js gzip | 921 B | 921 B | ✓ |
| image-HASH.js gzip | 5.59 kB | 5.59 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 2.33 kB | 2.33 kB | ✓ |
| routerDirect..HASH.js gzip | 322 B | 322 B | ✓ |
| script-HASH.js gzip | 388 B | 388 B | ✓ |
| withRouter-HASH.js gzip | 317 B | 317 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 15.8 kB | 15.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js add/e2e-test-react-17 | Change |
| - | - | - | - |
| index.html gzip | 532 B | 532 B | ✓ |
| link.html gzip | 546 B | 546 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

```diff
@@ -1505,14 +1505,14 @@
           },
           [callbacks]
         );
-        if (false) {
-        }
         // We should ask to measure the Web Vitals after rendering completes so we
         // don't cause any hydration delay:
         _react.default.useEffect(function() {
           (0, _performanceRelayer).default(onPerfEntry);
           (0, _vitals).flushBufferedVitalsMetrics();
         }, []);
+        if (false) {
+        }
         return children;
       }
       // Dummy component that we render as a child of Root so that we can
```
</details>
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-444bd3ddf1f35edc.js"
+      src="/_next/static/chunks/main-ca8f6d6aba194701.js"
       defer=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-444bd3ddf1f35edc.js"
+      src="/_next/static/chunks/main-ca8f6d6aba194701.js"
       defer=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -19,7 +19,7 @@
       defer=""
     ></script>
     <script
-      src="/_next/static/chunks/main-444bd3ddf1f35edc.js"
+      src="/_next/static/chunks/main-ca8f6d6aba194701.js"
       defer=""
     ></script>
     <script
```
</details>
</details>
Commit: d4ee34304123d161af1c4689175782ee5a7e9182

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**